napalm_eos/eos.py

```diff
@@ -1201,3 +1201,162 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
+                    try:
+                        peer_info[key] = int(peer_info[key])
+                    except Exception:
+                        # Defaulting the value to 0 in case of an exception
+                        peer_info[key] = 0
+
+                # Determining a few other fields in the final peer_info
+                peer_info['up'] = (
+                    True if peer_info['up'] == "up" else False)
+                peer_info['local_address_configured'] = (
+                    True if peer_info['local_address'] else False)
+                peer_info['multihop'] = (
+                    False if peer_info['multihop'] == 0 or
+                    peer_info['multihop'] == '' else True)
+
+                # TODO: The below fields need to be retrieved
+                # Currently defaulting their values to False or 0
+                peer_info['multipath'] = False
+                peer_info['remove_private_as'] = False
+                peer_info['suppress_4byte_as'] = False
+                peer_info['local_as_prepend'] = False
+                peer_info['flap_count'] = 0
+
+                # Appending peer_info to final data structure
+                bgp_detail_info[int(peer_info['remote_as'])].append(peer_info)
+
+        else:
+            # Using preset template to extract peer info
+            peer_info = (
+                napalm_base.helpers.textfsm_extractor(
+                    self, 'bgp_detail', raw_output)[0])
+
+            # Issuing command to get number of accepted prefixes for peer
+            if peer_ver == 6:
+                acc_prefix_command = (
+                    'show ipv6 bgp summary | include %s' % item[0])
+            else:
+                acc_prefix_command = (
+                    'show ip bgp summary | include %s' % item[0])
+
+            accepted_prefix_output = (
+                self.device.run_commands([acc_prefix_command],
+                                         encoding='text')[0]['output'])
+
+            accepted_prefix_count = accepted_prefix_output.split()[-1]
+
+            # Converting certain fields into int
+            for key in int_fields:
+                try:
+                    peer_info[key] = int(peer_info[key])
+                except Exception:
+                    # Defaulting the value to 0 in case of an exception
+                    peer_info[key] = 0
+
+            # Determining a few other fields in the final peer_info
+            peer_info['up'] = (
+                True if peer_info['up'] == "up" else False)
+            peer_info['local_address_configured'] = (
+                True if peer_info['local_address'] else False)
+            peer_info['multihop'] = (
+                False if peer_info['multihop'] == 0 or
+                peer_info['multihop'] == '' else True)
+
+            # TODO: The below fields need to be retrieved
+            # Currently defaulting their values to False or 0
+            peer_info['multipath'] = False
```

Comment:
Here you should follow the DRY principles. Especially when defining some defaults.
You have in a couple of places pretty much the same parts of code.